### PR TITLE
Add: Tech War Factory

### PR DESCRIPTION
**Pull request by Jundiyy**
Unfinished ZH Tech building, brought over from NProject, fully working.
Still needs a commandset, but will do this once we have added command buttons to the unused units.

Also needs the cameo pics added to ZH files. I will upload them here as not sure how we will do this.

### CODE REVIEW

**Comment by Jundiyy** [comment]
[SMUserInterface512_003.zip](https://github.com/xezon/GeneralsGamePatch/files/7448948/SMUserInterface512_003.zip)

Tech War Factory pic can be seen on the bottom line, second to the right.

[MDUserInterface512.zip](https://github.com/xezon/GeneralsGamePatch/files/7448949/MDUserInterface512.zip)

The code used in NProject is this, of course we would need to change this depending how we implement the cameo.
`MappedImage SSTechFact_L
  Texture = SMUserInterface512_003.tga
  TextureWidth = 512
  TextureHeight = 512
  Coords = Left:245 Top:393 Right:365 Bottom:489
  Status = NONE
End`

File: Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini [hunk 1]
```diff
@@ -1197,3 +1197,218 @@ Object TechRadioStation
...
+
+  ; *** ART Parameters ***
+
+  SelectPortrait = SSTechFact_L
+  ButtonImage    = SSTechFact_L
+
+  ; ========================= Main Model ===============================
+  Draw = W3DModelDraw ModuleTag_01
+    OkToChangeModelColor = Yes
+    IgnoreConditionStates = SNOW NIGHT
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini]: This line is superfluous here.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini [hunk 2]
```diff
@@ -1197,3 +1197,218 @@ Object TechRadioStation
...
+      Model            = ZBFactory_E
+    End
+    AliasConditionState = RUBBLE CAPTURED
+
+  End
+
+  ; ========================= Flag Model ===============================
+  Draw = W3DModelDraw ModuleTag_02
+    OkToChangeModelColor = Yes
+    IgnoreConditionStates = SNOW NIGHT DAMAGED REALLYDAMAGED
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini]: Same.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini [hunk 3]
```diff
@@ -1197,3 +1197,218 @@ Object TechRadioStation
...
+    End
+    ConditionState  = RUBBLE
+      Model         = None
+    End
+
+  End
+
+  ; ======================== Factory Door ==============================
+  Draw = W3DModelDraw ModuleTag_03
+    IgnoreConditionStates = SNOW NIGHT DAMAGED REALLYDAMAGED
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini]: Same...

File: Patch104pZH/GameFilesEdited/Data/INI/CommandSet.ini [hunk 4]
```diff
@@ -5125,3 +5125,21 @@ CommandSet Boss_ChinaNuclearMissileCommandSetUpgrade
...
+  ;5  = 
+  ;6  = 
+  ;7  = 
+  ;8  = 
+  ;9  = 
+  ;10 = 
+  ;11 = 
+  ;12 = 
+  13 = Command_SetRallyPoint
+  ;14 =
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/CommandSet.ini]: We don't need all those commented lines.

Comment (Jundiyy) [reply Patch104pZH/GameFilesEdited/Data/INI/CommandSet.ini]: I was going to add the units in but was waiting for them to go through from Xezon. They will be filled up.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini (hunk 1 above)
Comment (Jundiyy) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini]: Which one? The `    IgnoreConditionStates = SNOW NIGHT`?

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini]: Yes, that is the line the comment is on.

Comment (Jundiyy) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini]: okay will remove, does it not need them?

Comment (commy2) [reply Patch104pZH/GameFilesEdited/Data/INI/Object/TechBuildings.ini]: Remove it. It does nothing and does not belong there.

**Comment by commy2** [comment]
- [ ] squash before merging
- [ ] needs ui text ("War Factory" as placeholder currently)

The ruin looks exactly like the badly damaged building, but without flag. We didn't like this for the Repair Bay, but I guess we do not have a rubble model, right?

**Review by commy2 — APPROVED** [review]

File: Patch104pZH/Scripts/MAKE_Patch104pZH.bat [hunk 5]
```diff
@@ -21,6 +21,7 @@ if "%ThisArg1%"=="build" (
     xcopy /y    "%GameFilesDir%\Art\Textures\exlaser3.dds"        "%GeneratedBigFilesUnpackedDir%\%BigName%\Art\Textures\"
     xcopy /y    "%GameFilesDir%\Art\Textures\gxmammoth_co.tga"    "%GeneratedBigFilesUnpackedDir%\%BigName%\Art\Textures\"
     xcopy /y    "%GameFilesDir%\Art\Textures\gxmammothalt_HI.tga" "%GeneratedBigFilesUnpackedDir%\%BigName%\Art\Textures\"
+    xcopy /y    "%GameFilesDir%\Art\Textures\SSTechWarFactory_L.tga" "%GeneratedBigFilesUnpackedDir%\%BigName%\Art\Textures\"
```
Comment (xezon) [inline Patch104pZH/Scripts/MAKE_Patch104pZH.bat:24]: The W3D Models and DDS textures are not added to the big file here, so they will not show correctly in game.

**Review by xezon — CHANGES_REQUESTED** [review]

File: Patch104pZH/GameFilesEdited/Data/INI/MappedImages/TextureSize_512/SSTechWarFactory.ini [hunk 6]
```diff
@@ -0,0 +1,7 @@
+MappedImage SSTechWarFactory_L
+  Texture = SSTechWarFactory_L.tga
+  TextureWidth = 128
+  TextureHeight = 128
+  Coords = Left:1 Top:1 Right:121 Bottom:97
+  Status = NONE
+End
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/MappedImages/TextureSize_512/SSTechWarFactory.ini:7]: Should go in  `Patch104pZH/GameFilesEdited/Data/INI/MappedImages/HandCreated/` folder now, with the rest.

**Pull request closed without merging**